### DISCUSSION TITLE
Performance issues

### DISCUSSION THREAD

**Discussion by CGenie** [minad/vertico#461] [Q&A]
Hello,

I'm experiencing performance issues when doing `org-schedule`. Running profiler, I get this result: https://paste.sr.ht/~cgenie/39f9b9c6eb8a148f3d17ebd9e85aacee67e03c7e

I'm using stock ubuntu mantic emacs, version 29.1
```
GNU Emacs 29.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.17.8) of 2023-09-02, modified by Debian
```

Any ideas?

**Comment by minad** [comment]
It seems that `redisplay` appears prominently in your profile. This means you likely have some inefficient mode line activated.

**Comment by CGenie** [comment]
Hm, I analyzed minor modes that are enabled and tried to turn them off one by one, but that didn't help in performance. Could you elaborate what you mean by "inefficient mode line". My current `mode-line-format` variable is set to:
```emacs-lisp
(#2=""
    (eldoc-mode-line-string
     (#1=" " eldoc-mode-line-string #1#))
    ("%e" mode-line-front-space
     (:propertize
      (#2# mode-line-mule-info mode-line-client mode-line-modified mode-line-remote)
      display
      (min-width
       (5.0)))
     mode-line-frame-identification mode-line-buffer-identification "   " mode-line-position
     (vc-mode vc-mode)
     "  " mode-line-modes ...))
```

Also, my story is that I used ubuntu 23.04 which didn't have emacs 29, so I used the `emacs-snapshot` ppa which had some development version of emacs 30.x. That one worked fast for me and I didn't get any glitches. Then I upgraded ubuntu and used the stock 29.1 and all of a sudden I get big slowdowns on `org-schedule` like described above, all that without changing my emacs config.

**Reply by minad** [reply]
The mode line is executed on every redisplay. If you have expensive mode line constructs, they will lead to a slow down. But it could also be something else, some other inefficiency during redisplay.

I am sorry, I cannot help you further with this. Can you reproduce the problem with only Vertico and Org, starting from emacs -Q? See also the steps described here (applies to Consult, but similarly to Vertico): https://github.com/minad/consult?tab=readme-ov-file#bug-reports.
